build/cloudbuild-automation/README.md

```diff
@@ -0,0 +1,67 @@
+
+#### Prerequisites
+- Make sure that you have [docker](https://docs.docker.com/engine/install/) installed on your machine.
+    - Also verify that you are able to run docker commands without the need to gain root permissions using sudo. You can follow the [post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to achieve this.
+- Set the following environment variables 
+    - `PROJECT_ID` with the GCP project ID in which you will submit cloud builds.
+    - `GOOGLE_APPLICATION_CREDENTIALS` with the path to where your GCP credentials are present, most likely, it will be something like `${HOME}/.config/gcloud/application_default_credentials.json`.
+    - You can set them by executing  
+        ```bash
+        export PROJECT_ID=<your-project-id>
+        # This is where the default credentials live, might be different in case your default location is something else
+        export GOOGLE_APPLICATION_CREDENTIALS=${HOME}/.config/gcloud/application_default_credentials.json
+        ```
+
+#### Repository Structure - Cloudbuild-Automation
+
+1. `tf/main.tf` - folder containing Terraform files that will create necessary resources in your GCP account. 
+2. `tf/setup-build-automation.sh` - convenience shell script that runs the terraform script to generate the resources in GCP account. 
+3. `builder-config.yaml` - configuration defining how to build the custom opentelemetry collector. 
+4. `cloudbuild.yaml` - the YAML file containing steps that need to be run in cloud build. The cloudbuild trigger uses this file to build the docker image for the opentelemetry custom collector.
+
+#### How to run this sample
+
+1. `cd` into the `cloudbuild-automation/tf`. 
+2. Modify the variables declared at the top in `setup-build-automation.sh` script to suit your needs. 
+3. Run the `setup-build-automation.sh` script. 
+    ```
+    sh setup-build-automation.sh
+    ```
+4. Upon running the script, terraform should initialize and will generate a few files in the current directory.
+5. **IMPORTANT:** When you run the script for the first time, you will notice that the script fails with an error asking you to connect the repository to your GCP account. You will have to resolve this error before continuing, look at the [Connecting your repository to GCP account](#connecting-your-repository-to-gcp-account) section. 
+6. After running the script successfuly and connecting your repository, you should have a Cloud Build trigger and an Artifact Registry configured for your selected project. You can verify this by looking these services in [Google Cloud Console](https://console.cloud.google.com).
```

Comment:
```suggestion
6. After running the script successfuly and connecting your repository, you will have a Cloud Build trigger and an Artifact Registry configured for your selected project. You can verify this by looking these services in [Google Cloud Console](https://console.cloud.google.com).
```
try to replace instances of `should` with a more definitive statement where possible